widgetssdk/src/main/java/com/glia/widgets/chat/adapter/ChatMessage.kt

```diff
@@ -0,0 +1,53 @@
+package com.glia.widgets.chat.adapter
+
+import androidx.annotation.VisibleForTesting
+import com.glia.widgets.chat.adapter.ChatMessage.Participant.Companion.toWidgetsType
+import com.glia.androidsdk.chat.ChatMessage as CoreChatMessage
+import org.json.JSONObject
+
+/**
+ * Represents a chat message in the Glia Widgets SDK.
+ *
+ * @property id The unique identifier of the message.
+ * @property content The content of the message.
+ * @property timestamp The timestamp of when the message was received by the server, in milliseconds.
+ * @property deliveredAt The date when the message was marked as read, or `null` if not marked yet.
+ * @property attachment The attachment associated with the message, or `null` if none exists.
+ * @property metadata Custom payloads for tailoring the visitor experience, or `null` if none exists.
+ * @property senderType The type of sender (visitor or operator).
+ */
+class ChatMessage(
```

Comment:
Maybe it makes sense to make this class `data` class to have `toString` overriden by default?
I'm also unable to understand if integrators need need whole class or having `metadata` is enough.

Reply:
Made it a data class ef470376e3f759d92b1eba9be9cfdf9fba107080

Reply:
As far as I informed, Custom Cards are designed to use metadata.
Since we're going to introduce some breaking changes with closing Core SDK from the integrators, I guess it's okay to introduce one more and don't care about the integrators who use additional properties from the core's ChatMessage.

Reply:
@DavDo Aligned the `CustomCardMessage` structure with iOS `MessageRenderer` in [this](https://github.com/salemove/android-sdk-widgets/pull/1272/commits/c086431bdd4868996932d24e1025d3c2679fc24b#diff-608ce7c204e3c9cb11ccaf7ca8faf27a4b542540428c8fe6ea2c2fd1c2a34ae9) commit.

However, I'm not sure what shall I use instead of `.getContent()` [here in ExampleCustomCardAdapter](https://github.com/salemove/android-sdk-widgets/pull/1272/commits/c086431bdd4868996932d24e1025d3c2679fc24b#diff-b189e005acf6a0a8372478e87e6f4c0d542e9ca44b751a2e990ccac5f3208e77R106). Maybe @Andrii-Horishnii-Glia can help. 🤔